### DISCUSSION TITLE
How could i !git clone this repo?

### DISCUSSION THREAD

**Discussion by matroxI** [DominikDoom/a1111-sd-webui-tagcomplete#30] [Q&A] [answered]
Hello, sorry for the question a bit off topic, but I can't understand as I'm not related to programming in any way. How can I copy repository files to stable-diffusion-webui using !git clone.
- %cd stable-diffusion-webui
- !git clone https://github.com/DominikDoom/a1111-sd-webui-tagcomplete . 

Returned with an error
`
fatal: destination path '.' already exists and is not an empty directory
`
I use google colab, so manually inserting files every time is very inconvenient. Please, help. Thanks in advance.
Once again, I apologize very much, I looked at a couple of solutions, but I didn’t understand anything, I couldn’t apply them.

**Comment by sgmklp** [comment]
![image](https://user-images.githubusercontent.com/74038035/197361358-991ed498-6b66-463b-ad47-188ff4b8613e.png)
Don't add `.` in the last.
在末尾不要加 `.`

**Reply by matroxI** [reply]
![image](https://user-images.githubusercontent.com/111098250/197361936-64a2f670-3cc5-4e47-800c-f33a1ff28ac5.png)
It is not working like this. If you don't use a dot, then it just adds the repository folder to stable-diffusion-webui. Adding a dot makes it so that only files from the repository are copied, without the project folder itself.
![image](https://user-images.githubusercontent.com/111098250/197362271-9dae8906-0b88-41ed-a331-ffc8d7da1918.png)
For clarity, I created a test folder and did a git clone with a dot

**Comment by sgmklp** [comment, marked as answer]
git init
git remote add tagcomplete https://github.com/DominikDoom/a1111-sd-webui-tagcomplete.git
git pull main --allow-unrelated-histories

**Reply by sgmklp** [reply]
If you want to clone to a non empty directory, do this

**Reply by matroxI** [reply]
> git init git remote add tagcomplete https://github.com/DominikDoom/a1111-sd-webui-tagcomplete.git git pull main --allow-unrelated-histories

![image](https://user-images.githubusercontent.com/111098250/197362636-d9751ad6-78ab-4ee6-9f37-1388a6a3999b.png)

**Reply by sgmklp** [reply]
git pull tagcomplete  main --allow-unrelated-histories
my fault

**Reply by matroxI** [reply]
Thank you a lot, that worked. You also need to define yourself in the system in order for everything to work.
So the final code looks like this:

!git config --global user.email "Email"
!git config --global user.name "Username"
!git init
!git remote add tagcomplete https://github.com/DominikDoom/a1111-sd-webui-tagcomplete.git
!git pull tagcomplete main --allow-unrelated-histories

**Comment by DominikDoom** [comment]
Just as a quick update for those looking here, installing this as an extension is now possible, which drastically simplifies this process.
See [Installing as an extension](https://github.com/DominikDoom/a1111-sd-webui-tagcomplete#as-an-extension-recommended).

Since extensions are in a separate folder, they can keep their own git repo without the workarounds mentioned here.
